gnosis/safe-react
# fix: remove redundant `PENDING_FAILED` status

## What it solves
Resolves #3352

## How this PR fixes it
The `PENDING_FAILED` status was seemingly added as a placeholder and was causing the issue that it was set `onError`, irregardless of whether the transaction was a signature or execution. It has now been removed and the 'local statuses' converted to a 'pending transactions' array that stores and array of `safeTxHash`es instead.

The coniditons of performing an off-chain signature now first take into account whether off-chain signing is possible: `!isFinalization && canSignOffChain`.

The choice to rename `isExecution` to `isFinalization` is included here as it is more declarative.

## How to test it
The `PENDING` status should be shown when a signature is requested/the transaction is executed. The button text should also represent whether the transaction needs to be confirmed or executed.

The `PENDING` statuses should be broadcast across windows/tabs. If you open two tabs/windows of the queue list and execute the transaction in one, it should show the updated status in the other.

This logic affects a lot of cases across all transactions. This has been tested locally across all of the following, but it is best to check them again:

On a 1/? Safe:

- [ ] Rejecting the creation of a transaction.
- [ ] Creating a transaction and executing it immediately.
- [ ] Creating a transaction and executing it, rejecting the request.
- [ ] Queueing a transaction and approving it by an owner on a 1/1 Safe.
  - [ ] Confirm and execute it by an owner.
  - [ ] Confirm and execute it by a third owner.
  - [ ] Confirm and execute it by a non-owner.

On a 2/? Safe:

- [ ] Create a transaction on a 2/1 Safe.
  - [ ] Confirm and execute it by an owner.
  - [ ] Confirm and execute it by a third owner.
  - [ ] Confirm and execute it by a non-owner.

Creating and executing rejection transactions on the above should also be tested.
